lib/tbd/psi.rb

```diff
@@ -1771,13 +1769,125 @@ def process(model = nil, argh = {})
       end
     end
 
+    # Fetch edge multipliers for subsurfaces, if applicable.
+    edges.values.each do |edge|
+      next     if edge.key?(:mult)                    # skip if already assigned
+      next unless edge.key?(:surfaces)
+      next unless edge.key?(:psi)
+      ok = false
+
+      edge[:psi].keys.each do |k|
+        break if ok
+
+        jamb = k.to_s.include?("jamb")
+        sill = k.to_s.include?("sill")
+        head = k.to_s.include?("head")
+        ok   = jamb || sill || head
+      end
+
+      next unless ok     # if OK, edge links subsurface(s) ... yet which one(s)?
+
+      edge[:surfaces].each do |id, surface|
+        next unless tbd[:surfaces].key?(id)    # look up parent (opaque) surface
+
+        [:windows, :doors, :skylights].each do |subtypes|
+          next unless tbd[:surfaces][id].key?(subtypes)
+
+          tbd[:surfaces][id][subtypes].each do |nom, sub|
+            next unless edge[:surfaces].key?(nom)
+            next unless sub[:mult] > 1
+
+            # An edge may be tagged with (potentially conflicting) multipliers.
+            # This is only possible if the edge links 2 subsurfaces, e.g. a
+            # shared jamb between window & door. By default, TBD tags common
+            # subsurface edges as (mild) "transitions" (i.e. PSI 0 W/K.m), so
+            # there would be no point in assigning an edge multiplier. Users
+            # can however reset an edge type via a TBD JSON input file (e.g.
+            # "joint" instead of "transition"). It would be a very odd choice,
+            # but TBD doesn't prohibit it. If linked subsurfaces have different
+            # multipliers (e.g. 2 vs 3), TBD tracks the highest value.
+            edge[:mult] = sub[:mult] unless edge.key?(:mult)
+            edge[:mult] = sub[:mult]     if sub[:mult] > edge[:mult]
+          end
+        end
+      end
+    end
+
+    # Unless a user has set the thermal bridge type of an individual edge via
+    # JSON input, reset any subsurface's head, sill or jamb edges as (mild)
+    # transitions in close proximity with other similar subsurface edges.
+    # User-set tolerance must be greater than default TBD tolerance (0.01 m).
+    # Both edges' origin and terminal vertices must be in close proximity. Edges
+    # of unhinged subsurfaces are ignored.
+    if argh[:sub_tol] > TOL
+      edges.each do |id, edge|
+        nb    = 0                            # linked subsurfaces (i.e. "holes")
+        match = false
+        next if edge.key?(:io_type)          # skip if set in JSON
```

Comment:
I might be missing something but are these keys in the JSON schema, https://github.com/rd2/tbd/blob/develop/tbd.schema.json?

I see a `type` property but no `io_type`

Reply:
Ultimately, they refer to the same. `io_type` holds a user-set edge [type](https://github.com/rd2/tbd/blob/420054cebfeac4cd1c17b626bbadc9d44b24e193/lib/tbd/psi.rb#L752), parsed from a TBD JSON input file, which would likely be different from the edge type TBD would normally identify from its ruleset. In the end, if `io_type` is set by the user for a given edge, TBD doesn't even try to tag the edge (see [here](https://github.com/rd2/tbd/blob/420054cebfeac4cd1c17b626bbadc9d44b24e193/lib/tbd/psi.rb#L1271)).

Reply:
Oh I see, so the schema in the repo is only for TBD JSON input (user facing)?  I probably knew that and forgot.  It might be good to make a separate schema for the internal or output models?  Mostly just documenting it for yourself in the future.

Reply:
Actually, the JSON schema is good for both input and output files. We have several examples where a TBD JSON output file is reused as a TBD JSON input file a second time around.

Reply:
Do you want to add the `io_type` or other properties to the [schema](https://github.com/rd2/tbd/blob/develop/tbd.schema.json) in this PR?